TypeScript ts(2307) error when importing BaseAdapter with moduleResolution: "nodenext"

When using TypeScript with modern module resolution options ("moduleResolution": "nodenext" or "node16"), importing BaseAdapter fails with a ts(2307) error.

This happens because nodenext requires deep import paths to either include a file extension (e.g., .js) or be explicitly defined in the package.json "exports" field.

https://github.com/felixmosh/bull-board/blob/master/packages/nestjs/src/bull-board.types.ts#L7

`import type { BaseAdapter } from '@bull-board/api/dist/queueAdapters/base';`

`Error (when running tsc):`

`node_modules/@bull-board/nestjs/dist/bull-board.types.d.ts:3:34 - error TS2307: Cannot find module '@bull-board/api/dist/queueAdapters/base' or its corresponding type declarations.`

**Proposed Solution**

This can be fixed by adding an explicit entry to the "exports" field in the @bull-board/api package.json as it was done with the other adapters.
https://github.com/felixmosh/bull-board/blob/master/packages/api/package.json#L31

```
{
  "exports": {
    ".": "./dist/index.js",
    "./bullMQAdapter": "./bullMQAdapter.js",
    "./bullAdapter": "./bullAdapter.js",
    "./baseAdapter": "./baseAdapter.d.ts",
    // ...other exports
  }
}
```
Note that the [file](https://github.com/felixmosh/bull-board/blob/master/packages/api/baseAdapter.d.ts#L1) that exports the BaseAdapter from the dist folder was already created but it is not being used.

This would allow to import the type without resorting to workarounds like manually appending the file to the import or setting "skipLibCheck": true.

--- a/packages/api/package.json
+++ b/packages/api/package.json
@@ -29,6 +29,7 @@
     "./typings/*": "./typings/*.d.ts",
     "./bullMQAdapter": "./bullMQAdapter.js",
     "./bullAdapter": "./bullAdapter.js",
+    "./baseAdapter": "./baseAdapter.d.ts",
     "./constants/statuses": "./dist/constants/statuses.js"
   },
   "scripts": {

--- a/packages/nestjs/src/bull-board.types.ts
+++ b/packages/nestjs/src/bull-board.types.ts
@@ -4,7 +4,7 @@ import type {
   IServerAdapter,
   QueueAdapterOptions,
 } from '@bull-board/api/typings/app';
-import type { BaseAdapter } from '@bull-board/api/dist/queueAdapters/base';
+import type { BaseAdapter } from '@bull-board/api/baseAdapter';
 import type { InjectionToken, ModuleMetadata, OptionalFactoryDependency } from '@nestjs/common';
 
 export type BullBoardInstance = ReturnType<typeof createBullBoard>;

